src/main/java/io/aiven/kafka/connect/gcs/config/FilenameTemplateValidator.java

```diff
@@ -1,126 +0,0 @@
-/*
- * Aiven Kafka GCS Connector
- * Copyright (c) 2019 Aiven Oy
- *
- * This program is free software: you can redistribute it and/or modify
- * it under the terms of the GNU Affero General Public License as
- * published by the Free Software Foundation, either version 3 of the
- * License, or (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU Affero General Public License for more details.
- *
- * You should have received a copy of the GNU Affero General Public License
- * along with this program.  If not, see <https://www.gnu.org/licenses/>.
- */
-
-package io.aiven.kafka.connect.gcs.config;
-
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-import java.util.stream.Collectors;
-
-import org.apache.kafka.common.config.ConfigDef;
-import org.apache.kafka.common.config.ConfigException;
-
-import io.aiven.kafka.connect.gcs.templating.Pair;
-import io.aiven.kafka.connect.gcs.templating.Template;
-import io.aiven.kafka.connect.gcs.templating.VariableTemplatePart.Parameter;
-
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableMap;
-import com.google.common.collect.ImmutableSet;
-
-final class FilenameTemplateValidator implements ConfigDef.Validator {
```

Comment:
I understand that this validation logic moved to (effectively) `GcsSinkConfig`'s constructor, but I'd ask you to keep in the explicit `FilenameTemplateValidator` as much of it as possible. The reason for this is that it's processed differently by Connect. Errors that happen in  `FilenameTemplateValidator` will be reported back to user immediately, synchronously. Errors that happen during connector's `start()` happen (well) on the connector start, in asynchronous manner. If we look at this through the connector creation, in the first case the connector won't be created, in the second, will be created and immediately fail with an error. From the developer experience point of view, the former is much more preferable.

Some validation require multiple config options to check, it's impossible to do this in a single validator. But if an a config options is checked in isolation, it's better to do this in a validator.